mmdet/models/detectors/base.py

```diff
@@ -155,21 +173,41 @@ def forward(self, img, img_metas, return_loss=True, **kwargs):
         else:
             return self.forward_test(img, img_metas, **kwargs)
 
-
     def forward_export(self, imgs):
         from torch.onnx.operators import shape_as_tensor
+        assert self.img_metas, "Error: forward_export should be called inside forward_export_context"
+
         img_shape = shape_as_tensor(imgs[0])
         imgs_per_gpu = int(imgs[0].size(0))
         assert imgs_per_gpu == 1
+        assert len(self.img_metas[0]) == imgs_per_gpu, f"self.img_metas={self.img_metas}"
         self.img_metas[0][0]['img_shape'] = img_shape[2:4]
+
         return self.simple_test(imgs[0], self.img_metas[0], postprocess=False)
 
-    def export(self, img, img_metas, export_name='', **kwargs):
+    @contextmanager
+    def forward_export_context(self, img_metas):
+        assert self.img_metas is None and self.forward_backup is None, "Error: one forward context inside another forward context"
+
+        self.img_metas = img_metas
+        self.forward_backup = self.forward
+        self.forward = partial(self.forward, return_loss=False, forward_export=True, img_metas=None)
```

Comment:
As I wrote in my [big comment below](https://github.com/openvinotoolkit/mmdetection/pull/73#issuecomment-707357700), it is required to avoid replacing of `forward` method. There were some cases when replacing of the `forward` method worked incorrectly with NNCF, since NNCF uses its own tricks with `forward` method.

If we replace `forward` by `partial` of the same `forward` with "hardwired" parameters it is a safe decision -- we do not care which `forward` method is substituted, and even if `forward` method was wrapped and substituted by somebody else. If during wrapping the keyword arguments are passed as `**kwargs` to the original `forward`, this trick will work in any case.